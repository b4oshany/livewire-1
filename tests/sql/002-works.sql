\unset ECHO
\i tests/test_setup.sql
\i tests/navassa_data.sql
/*******************************************************

Test Suite for LiveWire


*******************************************************/


SELECT plan(4);
SELECT lw_initialise('navassa_shadow',3450);

SELECT isnt_empty(
  'SELECT * FROM navassa_shadow.navassa_shadow',
  'the navassa_shadow table must not be empty.'
  );

SELECT is_empty(
  'SELECT * FROM navassa_shadow.__livewire',
  'the __livewire table must be empty.'
  );

SELECT is_empty(
  'SELECT * FROM navassa_shadow.__lines',
  'the __lines table must be empty.'
  );

SELECT is_empty(
  'SELECT * FROM navassa_shadow.__nodes',
  'the __nodes table must be empty.'
  );


-- Non Test functions follow:

select lw_addedgeparticipant('navassa_shadow','{
  "schemaname":"navassa",						 
  "tablename": "primary_ag",
  "primarykey":"lineid",
  "geomcolumn": "g",
  "feederid": "feedername",
  "phasecolumn": "phases",
  "phasemap":{"ABC":"ABC","AB":"AB","AC":"AC","BC":"BC","A":"A","B":"B","C":"C"}
}');

select lw_addedgeparticipant('navassa_shadow','{
  "schemaname":"navassa",						 
  "tablename": "primary_ug",
  "primarykey":"lineid",
  "geomcolumn": "g",
  "feederid": "feedername",
  "phasecolumn": "phases",
  "phasemap":{"ABC":"ABC","AB":"AB","AC":"AC","BC":"BC","A":"A","B":"B","C":"C"}
}');

select lw_addedgeparticipant('navassa_shadow','{
  "schemaname":"navassa",						 
  "tablename": "risers",
  "primarykey":"lineid",
  "geomcolumn": "g",
  "feederid": "feedername",
  "phasecolumn": "phases",
  "phasemap":{"ABC":"ABC","AB":"AB","AC":"AC","BC":"BC","A":"A","B":"B","C":"C"}
}');

select lw_addedgeparticipant('navassa_shadow','{
  "schemaname":"navassa",						 
  "tablename": "jumpers",
  "primarykey":"lineid",
  "geomcolumn": "g",
  "feederid": "feedername",
  "phasecolumn": "phases",
  "phasemap":{"ABC":"ABC","AB":"AB","AC":"AC","BC":"BC","A":"A","B":"B","C":"C"}
}');

select lw_addnodeparticipant('navassa_shadow', $${
  "schemaname":"navassa",						 
  "tablename": "substation_transformers",
  "primarykey":"device_id",
  "geomcolumn": "g",
  "feederid":"feedername",
  "sourcequery": "1=1",
  "blockquery": "status='CLOSED'",
  "phasecolumn": "phases",
  "phasemap":{"ABC":"ABC","AB":"AB","AC":"AC","BC":"BC","A":"A","B":"B","C":"C"}
    }$$);
	
select lw_addnodeparticipant('navassa_shadow', $${
  "schemaname":"navassa",						 
  "tablename": "isolating_devices",
  "primarykey":"device_id",
  "geomcolumn": "g",
  "feederid":"feedername",
  "sourcequery": "1=2",
  "blockquery": "status='OPEN'",
  "phasecolumn": "phases",
  "phasemap":{"ABC":"ABC","AB":"AB","AC":"AC","BC":"BC","A":"A","B":"B","C":"C"}
    }$$);

select lw_addnodeparticipant('navassa_shadow', $${
  "schemaname":"navassa",						 
  "tablename": "transformer_devices",
  "primarykey":"device_id",
  "geomcolumn": "g",
  "feederid":"feedername",
  "sourcequery": "1=2",
  "blockquery": "status='OPEN'",
  "phasecolumn": "phases",
  "phasemap":{"ABC":"ABC","AB":"AB","AC":"AC","BC":"BC","A":"A","B":"B","C":"C"}
    }$$);
  



--SET client_min_messages TO 'ERROR';
SELECT lw_traceall('navassa_shadow');
--RESET client_min_messages;
-- Non test functions end.
/*SELECT isnt_empty(
  'SELECT * FROM navassa_shadow.__livewire',
  'the __livewire table must not be empty.'
  );
*/


SELECT * FROM finish();

ROLLBACK;
