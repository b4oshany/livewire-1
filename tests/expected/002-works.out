\unset ECHO
psql:tests/navassa_queries.sql:82: NOTICE:  SOURCE: 1 | 1 of 7
psql:tests/navassa_queries.sql:82: NOTICE:  @ 0.033503 secs | Elapsed time is @ 0.06088 secs
psql:tests/navassa_queries.sql:82: NOTICE:  SOURCE: 2 | 2 of 7
psql:tests/navassa_queries.sql:82: NOTICE:  @ 0.052858 secs | Elapsed time is @ 0.113788 secs
psql:tests/navassa_queries.sql:82: NOTICE:  SOURCE: 3 | 3 of 7
psql:tests/navassa_queries.sql:82: NOTICE:  @ 0.059214 secs | Elapsed time is @ 0.173046 secs
psql:tests/navassa_queries.sql:82: NOTICE:  SOURCE: 4 | 4 of 7
psql:tests/navassa_queries.sql:82: NOTICE:  @ 0.023634 secs | Elapsed time is @ 0.196722 secs
psql:tests/navassa_queries.sql:82: NOTICE:  SOURCE: 5 | 5 of 7
psql:tests/navassa_queries.sql:82: NOTICE:  @ 0.039029 secs | Elapsed time is @ 0.23579 secs
psql:tests/navassa_queries.sql:82: NOTICE:  SOURCE: 6 | 6 of 7
psql:tests/navassa_queries.sql:82: NOTICE:  @ 0.018247 secs | Elapsed time is @ 0.254086 secs
psql:tests/navassa_queries.sql:82: NOTICE:  SOURCE: 7 | 7 of 7
psql:tests/navassa_queries.sql:82: NOTICE:  @ 0.0232 secs | Elapsed time is @ 0.277336 secs
1..2
not ok 1 - the __livewire table must be empty.
# Failed test 1: "the __livewire table must be empty."
#     Unexpected records:
#         ("{1,136,254,108,143}","{132,131,311,250}")
#         ("{1,136,254,172,354,45,215,28,208,174,141,114,145}","{132,131,130,129,158,159,17,18,19,20,319,258}")
#         ("{1,136,254,172,354,45,215,137,242,348,116,171}","{132,131,130,129,158,159,160,161,162,328,267}")
#         ("{1,136,254,172,354,45,215,137,115,186}","{132,131,130,129,158,159,160,320,259}")
#         ("{1,136,254,172,354,144,107,193}","{132,131,130,129,128,329,268}")
#         ("{1,136,254,172,354,144,176,241,105,204}","{132,131,130,129,128,127,126,339,278}")
#         ("{1,136,254,172,354,144,176,106,290}","{132,131,130,129,128,127,334,273}")
#         ("{1,136,254,172,354,45,215,28,208,174,141,247,71,325}","{132,131,130,129,158,159,17,18,19,20,21,327,266}")
#         ("{2,210,225,278,27,320,182,214,140,122,112,135}","{47,48,49,10,11,12,13,14,15,314,253}")
#         ("{2,210,225,278,250,253,166}","{47,48,49,50,51,212}")
#         ("{2,210,225,278,250,253,307,31,264,258,103,167}","{47,48,49,50,51,52,53,54,55,298,237}")
#         ("{2,210,225,278,250,253,307,19,311,177,120,346,337,20,189,333,224,157,138,175,127,61,179}","{47,48,49,50,51,52,209,167,168,169,170,171,172,173,174,175,176,177,178,179,233,222}")
#         ("{2,210,225,278,250,253,307,19,311,177,120,346,337,20,189,333,224,157,138,48,219}","{47,48,49,50,51,52,209,167,168,169,170,171,172,173,174,175,176,177,234,223}")
#         ("{2,210,225,278,27,320,182,214,140,122,338,113,223}","{47,48,49,10,11,12,13,14,15,16,322,261}")
#         ("{2,210,225,278,250,253,307,31,264,104,252}","{47,48,49,50,51,52,53,54,297,236}")
#         ("{2,210,225,278,27,320,182,26,309,123,109,266}","{47,48,49,10,11,12,7,8,9,316,255}")
#         ("{2,210,225,278,250,253,307,19,311,177,22,322,164,159,23,251,56,267}","{47,48,49,50,51,52,209,167,168,182,183,184,185,207,208,229,218}")
#         ("{2,210,225,278,27,320,182,214,111,268}","{47,48,49,10,11,12,13,310,249}")
#         ("{2,210,225,278,250,253,307,19,311,177,120,59,270}","{47,48,49,50,51,52,209,167,168,169,226,215}")
#         ("{2,210,225,278,250,117,282}","{47,48,49,50,301,240}")
#         ("{2,210,225,278,250,253,307,19,311,177,22,322,164,159,121,328,54,292}","{47,48,49,50,51,52,209,167,168,182,183,184,185,186,187,231,220}")
#         ("{2,210,225,278,250,253,307,19,311,60,293}","{47,48,49,50,51,52,209,167,225,214}")
#         ("{2,210,225,278,250,253,307,19,311,177,22,322,47,303}","{47,48,49,50,51,52,209,167,168,182,183,235,224}")
#         ("{2,210,225,278,250,253,307,19,311,177,22,322,164,21,244,57,308}","{47,48,49,50,51,52,209,167,168,182,183,184,180,181,228,217}")
#         ("{2,210,225,278,250,253,307,19,311,177,22,322,164,159,121,55,312}","{47,48,49,50,51,52,209,167,168,182,183,184,185,186,230,219}")
#         ("{2,210,225,278,250,253,307,19,311,177,120,346,337,20,189,333,224,53,329}","{47,48,49,50,51,52,209,167,168,169,170,171,172,173,174,175,232,221}")
#         ("{2,210,225,278,250,253,307,19,311,177,120,346,337,58,353}","{47,48,49,50,51,52,209,167,168,169,170,171,227,216}")
#         ("{2,210,225,278,27,320,182,26,309,110,355}","{47,48,49,10,11,12,7,8,309,248}")
#         ("{3,132,191,161,277,339,294,165,255,239,32,233,216,156,226,228,335,13,357,269,336,347,134,6}","{57,58,59,60,61,62,63,64,65,66,67,40,36,37,38,143,144,145,146,147,148,149,150}")
#         ("{3,132,191,161,277,339,294,165,77,119}","{57,58,59,60,61,62,63,357,296}")
#         ("{3,132,191,161,277,339,294,165,255,239,32,233,216,156,226,228,335,81,124}","{57,58,59,60,61,62,63,64,65,66,67,40,36,37,38,143,343,282}")
#         ("{3,132,191,161,277,339,294,165,255,239,32,233,216,156,30,305,235,192,66,131}","{57,58,59,60,61,62,63,64,65,66,67,40,36,43,44,45,46,345,284}")
#         ("{3,132,191,161,277,339,294,165,255,239,32,233,216,156,30,305,235,192,163,207,43,295,275,64,152}","{57,58,59,60,61,62,63,64,65,66,67,40,36,43,44,45,46,122,123,133,134,135,340,279}")
#         ("{3,132,191,161,277,339,294,165,255,239,32,233,216,156,226,228,335,13,357,269,336,160}","{57,58,59,60,61,62,63,64,65,66,67,40,36,37,38,143,144,145,146,147,22}")
#         ("{3,132,191,161,277,339,294,165,255,239,32,233,216,156,226,79,183}","{57,58,59,60,61,62,63,64,65,66,67,40,36,37,349,288}")
#         ("{3,132,191,161,277,339,294,165,255,239,32,233,216,156,226,228,335,13,357,269,197}","{57,58,59,60,61,62,63,64,65,66,67,40,36,37,38,143,144,145,146,163}")
#         ("{3,132,191,161,277,339,294,165,255,239,32,233,29,259,327,149,243,76,200}","{57,58,59,60,61,62,63,64,65,66,67,25,26,27,28,29,353,292}")
#         ("{3,132,191,161,277,339,294,165,255,239,32,233,29,259,327,149,18,301,287,288,198,73,201}","{57,58,59,60,61,62,63,64,65,66,67,25,26,27,28,31,32,33,34,35,347,286}")
#         ("{3,132,191,161,277,339,294,165,255,239,32,233,216,156,226,228,335,13,357,269,336,347,15,237,212,342,187,313,356,126,46,229}","{57,58,59,60,61,62,63,64,65,66,67,40,36,37,38,143,144,145,146,147,148,153,154,138,139,140,141,75,76,77,213}")
#         ("{3,132,191,161,277,339,294,165,255,239,32,233,216,156,30,305,235,192,163,274,133,78,234}","{57,58,59,60,61,62,63,64,65,66,67,40,36,43,44,45,46,122,156,157,342,281}")
#         ("{3,132,191,161,277,339,294,165,255,239,32,233,29,259,327,62,265}","{57,58,59,60,61,62,63,64,65,66,67,25,26,27,355,294}")
#         ("{3,132,191,161,277,339,294,165,255,239,32,233,216,156,30,305,235,192,163,207,43,295,275,218,199,65,271}","{57,58,59,60,61,62,63,64,65,66,67,40,36,43,44,45,46,122,123,133,134,135,136,137,333,272}")
#         ("{3,132,191,161,277,339,294,165,255,239,32,233,29,259,327,149,18,301,74,289}","{57,58,59,60,61,62,63,64,65,66,67,25,26,27,28,31,32,352,291}")
#         ("{3,132,191,161,277,339,294,165,255,239,32,233,216,156,226,228,80,330}","{57,58,59,60,61,62,63,64,65,66,67,40,36,37,38,346,285}")
#         ("{3,132,191,161,277,339,294,165,255,239,32,233,29,259,327,149,243,146,75,340}","{57,58,59,60,61,62,63,64,65,66,67,25,26,27,28,29,30,350,289}")
#         ("{3,132,191,161,277,339,294,165,255,239,32,233,216,156,30,305,235,192,163,207,43,295,275,218,118,343}","{57,58,59,60,61,62,63,64,65,66,67,40,36,43,44,45,46,122,123,133,134,135,136,337,276}")
#         ("{3,132,191,161,277,339,294,165,255,239,32,233,216,156,30,305,235,192,163,207,43,295,63,349}","{57,58,59,60,61,62,63,64,65,66,67,40,36,43,44,45,46,122,123,133,134,341,280}")
#         ("{4,237,212,342,187,313,44,213,88,148}","{155,138,139,140,141,142,164,331,270}")
#         ("{4,237,212,342,187,313,38,153,85,169}","{155,138,139,140,141,103,104,332,271}")
#         ("{4,237,212,342,187,313,356,126,46,229}","{155,138,139,140,141,75,76,77,213}")
#         ("{4,237,212,342,187,313,356,126,94,283}","{155,138,139,140,141,75,76,313,252}")
#         ("{4,237,212,342,187,87,314}","{155,138,139,140,321,260}")
#         ("{4,237,212,342,187,313,356,33,350,90,323}","{155,138,139,140,141,75,68,69,323,262}")
#         ("{4,237,212,342,187,313,356,33,350,324,89,334}","{155,138,139,140,141,75,68,69,70,326,265}")
#         ("{4,237,212,342,86,345}","{155,138,139,330,269}")
#         ("{5,236,162,256,42,139,194,41,280,261,125,206,100,142}","{88,89,90,111,112,113,114,115,116,117,118,300,239}")
#         ("{5,236,162,40,205,358,37,184,36,248,217,129,68,147}","{88,89,105,106,107,97,98,99,100,101,102,304,243}")
#         ("{5,236,162,256,302,35,221,97,154}","{88,89,90,91,92,93,302,241}")
#         ("{5,236,162,40,205,358,39,195,96,168}","{88,89,105,106,107,108,109,324,263}")
#         ("{5,236,162,256,42,139,194,101,170}","{88,89,90,111,112,113,317,256}")
#         ("{5,236,162,40,205,358,37,184,240,310,318,304,93,173}","{88,89,105,106,107,97,98,71,72,73,74,308,247}")
#         ("{5,236,102,230}","{88,325,264}")
#         ("{5,236,162,256,42,139,194,41,280,261,70,232}","{88,89,90,111,112,113,114,115,116,303,242}")
#         ("{5,236,162,40,205,358,39,195,155,95,257}","{88,89,105,106,107,108,109,110,318,257}")
#         ("{5,236,162,256,42,139,194,41,280,261,125,206,298,286,69,262}","{88,89,90,111,112,113,114,115,116,117,118,119,120,299,238}")
#         ("{5,236,162,40,205,358,37,184,240,92,263}","{88,89,105,106,107,97,98,71,312,251}")
#         ("{5,236,162,40,205,358,37,184,36,248,217,67,276}","{88,89,105,106,107,97,98,99,100,101,307,246}")
#         ("{5,236,162,40,205,358,37,184,240,310,91,284}","{88,89,105,106,107,97,98,71,72,305,244}")
#         ("{5,236,162,256,42,139,194,41,280,99,299}","{88,89,90,111,112,113,114,115,306,245}")
#         ("{5,236,162,256,302,98,332}","{88,89,90,91,315,254}")
#         ("{6,134,347,336,269,357,24,185,300,83,128}","{150,149,148,147,146,3,4,41,335,274}")
#         ("{6,134,347,336,160}","{150,149,148,22}")
#         ("{6,134,347,336,269,197}","{150,149,148,147,163}")
#         ("{6,134,347,336,269,357,24,185,300,222,84,202}","{150,149,148,147,146,3,4,41,42,338,277}")
#         ("{6,134,347,336,269,357,24,185,25,296,82,321}","{150,149,148,147,146,3,4,5,6,336,275}")
#         ("{7,34,211,272,49,150}","{79,80,81,356,295}")
#         ("{7,34,211,272,273,151,351,331,181,52,158}","{79,80,81,82,83,84,85,78,348,287}")
#         ("{7,34,211,272,273,151,351,331,203,72,196}","{79,80,81,82,83,84,85,86,344,283}")
#         ("{7,34,211,272,273,151,351,319,344,51,245}","{79,80,81,82,83,84,23,24,351,290}")
#         ("{7,34,211,272,273,151,50,260}","{79,80,81,82,83,354,293}")
#         ("{7,34,211,272,273,151,351,331,203,17,316}","{79,80,81,82,83,84,85,86,87,211}")
ok 2 - the __livewire table must not be empty.
# Looks like you failed 1 test of 2
